tools/sggolangcilint/golangci.yml

```diff
@@ -160,120 +175,113 @@ linters:
-
-  gofumpt:
-    extra-rules: true
-
-  gosec:
-    excludes:
-      # Flags for potentially-unsafe casting of ints. Prone to lots of false positives.
-      - G115
-
-  lll:
-    line-length: 120
-    tab-width: 1
-
-  misspell:
-    locale: US
-    ignore-words:
-      - cancelled
-      - cancelling
-      - analyses # allow analyses as a plural form of analysis
-
-  prealloc:
-    simple: true
-    range-loops: true
-    for-loops: true
-
-  protogetter:
-    skip-any-generated: true
-
-  staticcheck:
-    checks: [all]
-
-  stylecheck:
-    checks: [all]
-
-  errorlint:
-    # Don't enforce %w to minimize implicit interface leakage.
-    # https://github.com/dwmkerr/hacker-laws#hyrums-law-the-law-of-implicit-interfaces
-    errorf: false
-
-  gomoddirectives:
-    replace-local: true
+  settings:
+    errorlint:
+      errorf: false
+
+    gomoddirectives:
+      replace-local: true
+
+    gomodguard:
+      blocked:
+        modules:
+          - github.com/sirupsen/logrus:
+              recommendations:
+                - go.uber.org/zap
+              reason: Use Zap for logging.
+          - github.com/rs/zerolog:
+              recommendations:
+                - go.uber.org/zap
+              reason: Use Zap for logging.
+          - github.com/pkg/errors:
+              recommendations:
+                - errors
+                - fmt
+              reason: Use the standard library error packages.
+          - golang.org/x/xerrors:
+              recommendations:
+                - errors
+                - fmt
+              reason: Use the standard library error packages.
+          - github.com/golang/protobuf:
+              recommendations:
+                - google.golang.org/protobuf
+              reason: The protobuf v1 module is deprecated.
+          - github.com/stretchr/testify:
+              recommendations:
+                - gotest.tools/v3
+              reason: More reliably supports protobuf messages.
+
+    gosec:
+      excludes:
+        - G115
+
+    lll:
+      line-length: 120
+      tab-width: 1
+
+    misspell:
+      locale: US
+      ignore-rules:
+        - cancelled
+        - cancelling
+        - analyses
+
+    prealloc:
+      simple: true
+      range-loops: true
+      for-loops: true
+
+    protogetter:
+      skip-any-generated: true
+
+    staticcheck:
+      checks:
+        - all
+
+  exclusions:
+    generated: lax
+    presets:
+      - comments
+      - common-false-positives
+      - legacy
+      - std-error-handling
+    paths:
+      - .git
+      - node_modules
+      - third_party$
+      - builtin$
+      - examples$
+
+formatters:
+  enable:
+    - gci
+    - goimports
+    - gofumpt
+    - golines
+
+  settings:
+    gofumpt:
+      extra-rules: true
+    golines:
+      max-len: 120
+
+  exclusions:
+    generated: lax
+    paths:
+      - .git
+      - node_modules
+      - third_party$
+      - builtin$
+      - examples$
```

Comment:
I'm noticing that `golangci-lint fmt` wants to format files much more heavily if this config resides inside the `.golangci.yml` (I suspect it uses default `max-len: 100`). If the formatters config is broken out into its own yaml file, it seems to behave more consistently with `max-len: 120`.